publication-rest/src/main/java/no/unit/nva/publication/update/PublishingRequestResolver.java

```diff
@@ -89,9 +89,11 @@ private void handlePublishingRequest(Publication oldImage, Publication newImage)
         var pendingPublishingRequests = fetchPendingPublishingRequestsForUserInstitution(oldImage);
         if (pendingPublishingRequests.isEmpty()) {
             createPublishingRequestOnFileUpdate(oldImage, newImage);
+            return;
         }
         if (thereAreNoFiles(newImage)) {
             autoCompletePendingPublishingRequestsIfNeeded(newImage, pendingPublishingRequests);
+            return;
         }
         if (updateHasFileChanges(oldImage, newImage)) {
             updateFilesForApproval(oldImage, newImage, pendingPublishingRequests);
```

Comment:
When removing unpublished files, both second and third conditions are true. Second condition approval publishing request. Third one updates files for approval (removes them) and updates completing publishing request to pending. Probably we need to restrict COMPLETED -> PENDING transition.